CMakeLists.txt

```diff
@@ -1,56 +1,186 @@
-        src/rpc/*.h src/rpc/*.cpp
+
+if (${DISABLE_BENCH})
+    set(OPTIONS "${OPTIONS} --disable-bench")
+endif()
+
+string(TOLOWER ${CMAKE_SYSTEM_NAME}-${CMAKE_SYSTEM_PROCESSOR} ARCH)
+
+CPMAddPackage(
+    NAME OpenSSL
+    GITHUB_REPOSITORY openssl/openssl
+    VERSION 1.0.2
+    GIT_TAG OpenSSL_1_0_2r
+    DOWNLOAD_ONLY TRUE
+)
+
+if(OpenSSL_ADDED)
+    ExternalProject_Add(OpenSSL
+        PREFIX openssl
+        SOURCE_DIR ${OpenSSL_SOURCE_DIR}
+        CONFIGURE_COMMAND ${OpenSSL_SOURCE_DIR}/Configure ${ARCH} no-shared no-dso no-engines -fPIC --prefix=<INSTALL_DIR>
+        BUILD_IN_SOURCE 1
+    )
+    set(DEPENDS ${DEPENDS} OpenSSL)
+    ExternalProject_Get_Property(OpenSSL INSTALL_DIR)
+    set(LDFLAGS "${LDFLAGS} -L${INSTALL_DIR}/lib")
+    set(CPPFLAGS "${CPPFLAGS} -I${INSTALL_DIR}/include")
+    set(OPENSSL_CPPFLAGS "CPPFLAGS=-I${INSTALL_DIR}/include")
+    set(OPENSSL_LDFLAGS "LDFLAGS=-L${INSTALL_DIR}/lib")
+endif(OpenSSL_ADDED)
+
+CPMAddPackage(
+    NAME Libevent
+    GITHUB_REPOSITORY libevent/libevent
+    VERSION 2.1.8
+    GIT_TAG release-2.1.8-stable
+    DOWNLOAD_ONLY TRUE
+)
+
+if(Libevent_ADDED)
+    ExternalProject_Add(Libevent
+        PREFIX libevent
+        DEPENDS ${DEPENDS}
+        SOURCE_DIR ${Libevent_SOURCE_DIR}
+        CONFIGURE_COMMAND ${Libevent_SOURCE_DIR}/autogen.sh
+            && ${Libevent_SOURCE_DIR}/configure ${OPENSSL_CPPFLAGS} --enable-cxx --disable-shared --with-pic ${OPENSSL_LDFLAGS} --prefix=<INSTALL_DIR>
+        BUILD_IN_SOURCE 1
     )
-list(FILTER sources EXCLUDE REGEX "src/bitcoin*.cpp$")
-
-include_directories(${Boost_INCLUDE_DIRS}
-        build/bdb/include
-        build/libevent/include
-        build/openssl/include
-        src/support/allocators
-        src/support
-        src/rpc
-        src/policy
-        src/wallet src/script
-        src/leveldb/helpers/memenv
-        src/leveldb/include
-        src/config
-        src/crypto
-        src/compat
-        src/obj
-        src/univalue/include
-        src/secp256k1/include
-        src/
+    set(DEPENDS ${DEPENDS} Libevent)
+    ExternalProject_Get_Property(Libevent INSTALL_DIR)
+    set(LDFLAGS "${LDFLAGS} -L${INSTALL_DIR}/lib")
+    set(CPPFLAGS "${CPPFLAGS} -I${INSTALL_DIR}/include")
+endif(Libevent_ADDED)
+
+if(NOT ${DISABLE_WALLET})
+    CPMAddPackage(
+        NAME BerkeleyDB
+        VERSION 4.8.30
+        URL https://download.oracle.com/berkeley-db/db-4.8.30.NC.zip
+        URL_HASH SHA256=43ecd76886992ea416fdadc54b7f2b83ef249d9a6964bd07708ccae42d0226ce
+        DOWNLOAD_ONLY TRUE
+    )
+
+    if(NOT ${BerkeleyDB_VERSION} VERSION_LESS "5.0")
+        set(OPTIONS "${OPTIONS} --with-incompatible-bdb")
+    endif()
+
+    if(BerkeleyDB_ADDED)
+        ExternalProject_Add(BerkeleyDB
+            PREFIX bdb
+            SOURCE_DIR ${BerkeleyDB_SOURCE_DIR}
+            PATCH_COMMAND patch -Nd ${BerkeleyDB_SOURCE_DIR}/dbinc -i ${CMAKE_CURRENT_SOURCE_DIR}/depends/patches/db/atomic.patch || true
```

Comment:
The "depends" system uses `sed` instead of `patch` to achieve this. It doesn't need the extra file. The `sed` approach seems superior to me.